### PR TITLE
Create Design 📗  and Views For Application 🎨 

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR I present Implementation Design 📗  and Views For Application 🎨 :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- All linters 🔍 should pass.
- Use GitFlow 🌊.
- Implement the design from the sneak peek wireframes ⛓️ 
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- Implement these wireframes [print1](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_root_users.png), [print2](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_page.png), [print3](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png), [print4](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png)

## Achievements
- Setup ruby workflow and linters 🔍. ✅ 
- use GitFlow 🌊.✅
- Implement the design from the sneak peek wireframes ⛓️ ✅
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md). ✅
- Implement these wireframes [print1](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_root_users.png), [print2](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_page.png), [print3](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png), [print4](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png) ✅

## Conclusion
**I look forward to hearing 👂  from you The CR in terms of what can be better implemented 👨‍💻  for the project.**

**Thankyou!.**  🙇

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,27 @@
...
+  
+  <section class="bio_section">
+    <h1 class="bio_header">Bio</h1>
+    <p class="bio_paragraph">
+      <%= @user.bio %>
+    </p>
+  </section>
+  
+  <section class="posts_section">
+    <% @user.posts.order(created_at: :asc).limit(3).each_with_index do |post, i| %>
```
Comment (TedLivist) [inline app/views/users/show.html.erb]: - [ ] The posts that are rendered on the user show page should be the ones gotten from the `recent_posts` method in the User model. Kindly do that.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,5 +1,13 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+    @comments = Comment.where(post_id: @user.posts.pluck(:id))
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [ ] The comments rendered on the post index page should be the ones gotten from the `recent_comments` method in the Post model. Kindly use that.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,27 @@
...
+      <div class="post">
+        <span class="title_content">
+          <h2 class="post_title">Post #<%=i + 1%></h2>
```
Comment (TedLivist) [inline app/views/users/show.html.erb]: - [ ] A part of the post, preferably the title, on the user show page should be a link that takes the user to the post. Kindly add this link.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @gisachris,

Good job so far!
✔️ Nice implementation of page layouts
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
@TedLivist thank you for the insight into the changes to make.  I will implement them immediately

**Comment by gisachris** [comment]
## IN This PR I present the Implementation of required changes♻️:

Below are some of The requested changes by a past Reviewer 👨‍💻.
## Requested changes ♻️ 
- use object method `recent_posts` for displaying recent user posts [check here for full reviewer remarks](https://github.com/gisachris/blogTastic/pull/6#discussion_r1311475409)
- use object method `recent_comments` for displaying recent user comments on posts [check here for full reviewer remarks](https://github.com/gisachris/blogTastic/pull/6#discussion_r1311477090)
- On the `user show page` i should introduce a link to the post details via the `title` of the `post` [check here for full reviewer remarks](https://github.com/gisachris/blogTastic/pull/6#discussion_r1311491622)

## Achievements
- use object method `recent_posts` for displaying recent user posts ✅
- use object method `recent_comments` for displaying recent user comments on posts ✅
- On the `user show page` i should introduce a link to the post details via the `title` of the `post` ✅ 

## Conclusion
**I look forward to hearing 👂  from you The CR in terms of what can be better implemented 👨‍💻  for the project.**

**Thankyou!.**  🙇

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @gisachris 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added views associated with the existing models and controllers 📌 
✔️ Good commit history
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**